editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/restconfiguration/RestConfigEditor.java

```diff
@@ -0,0 +1,676 @@
+import org.eclipse.jface.viewers.ISelectionChangedListener;
+import org.eclipse.jface.viewers.ISelectionProvider;
+import org.eclipse.jface.viewers.LabelProvider;
+import org.eclipse.jface.viewers.ListViewer;
+import org.eclipse.jface.viewers.SelectionChangedEvent;
+import org.eclipse.jface.viewers.StructuredSelection;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Combo;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Listener;
+import org.eclipse.swt.widgets.MessageBox;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.swt.widgets.ToolBar;
+import org.eclipse.ui.IEditorInput;
+import org.eclipse.ui.IEditorSite;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.forms.widgets.FormToolkit;
+import org.eclipse.ui.forms.widgets.ScrolledForm;
+import org.eclipse.ui.forms.widgets.Section;
+import org.eclipse.ui.part.EditorPart;
+import org.fusesource.ide.camel.editor.CamelEditor;
+import org.fusesource.ide.camel.editor.internal.CamelEditorUIActivator;
+import org.fusesource.ide.camel.editor.internal.UIMessages;
+import org.fusesource.ide.camel.model.service.core.model.CamelBasicModelElement;
+import org.fusesource.ide.camel.model.service.core.model.CamelFile;
+import org.fusesource.ide.camel.model.service.core.model.ICamelModelListener;
+import org.fusesource.ide.foundation.core.util.Strings;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+
+/**
+ * @author bfitzpat
+ */
+public class RestConfigEditor extends EditorPart implements ICamelModelListener, ISelectionProvider {
+
+	private Map<String, Color> colorMap;
+	private CamelEditor parentEditor;
+	private Composite parent;
+	private ScrolledForm form;
+	private FormToolkit toolkit;
+	private Map<String, List<Object>> model;
+	private ImageRegistry mImageRegistry;	
+	private ListenerList<ISelectionChangedListener> listeners = new ListenerList<>();
+	private Object selection;
+	private Control selectedControl;
+	private Combo componentCombo;
+	private Text contextPathText;
+	private Text portText;
+	private Combo bindingModeCombo;
+	private Text hostText;
+	private Composite restOpsSection;
+	private ListViewer restList;
+
+	/**
+	 *
+	 * @param parentEditor
+	 */
+	public RestConfigEditor(CamelEditor parentEditor) {
+		this.parentEditor = parentEditor;
+	}
+
+	@Override
+	public void doSave(IProgressMonitor monitor) {
+		this.parentEditor.doSave(monitor);
+	}
+
+	@Override
+	public void doSaveAs() {
+		this.parentEditor.doSaveAs();
+	}
+
+	@Override
+	public void init(IEditorSite editorSite, IEditorInput input)
+			throws PartInitException {
+		setSite(editorSite);
+		setInput(input);
+//		getSite().setSelectionProvider(this);
```

Comment:
deactivating the selection completely is then providing very little information on the rest configuration. Why not for a very first iteration as the editor is hidden by default  anyway.
Would need to update the properties view to not have the previous element selection still displayed (for instance element of the selected element in the design tab before going to Rest tab)